src/tests/ftest/deployment/network_failure.py

```diff
@@ -0,0 +1,398 @@
+        client node, engine, etc. are still working).
+        5. Run IOR with oclass SX.
+        6. Verify that IOR finishes successfully.
+        7. Verify that the container Health is HEALTHY.
+        8. To further verify that the pool isn’t affected, create a new container on the
+        pool and run IOR.
+        9. To clean up, bring up the network interface.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=deployment,network_failure
+        :avocado: tags=network_failure_isolation
+        """
+        # 1. Determine the two ranks to create the pool and an interface to take down.
+        # We'll create a pool on rank 0 and the other rank that's on the same node. Find
+        # hostname of rank 0.
+        rank_0_ip = None
+        output = self.get_dmg_command().system_query()
+        members = output["response"]["members"]
+        for member in members:
+            if member["rank"] == 0:
+                rank_0_ip = member["addr"].split(":")[0]
+                break
+        self.log.info("rank 0 IP = %s", rank_0_ip)
+
+        # Find the other rank that's on the same node as in rank 0. Call it rank_r.
+        rank_r = None
+        for member in members:
+            if member["addr"].split(":")[0] == rank_0_ip and member["rank"] != 0:
+                rank_r = member["rank"]
+                break
+        self.log.info("rank_r = %s", rank_r)
+
+        # Find the hostname that's different from rank_0_ip. We'll take down the
+        # interface on it.
+        # dmg system query output gives IP address, but run_pcmd doesn't work with IP in
+        # CI. In addition, in Aurora, it's easier to determine where to run the ip link
+        # command if we know the hostname, so create the IP - hostname mapping.
+        ip_to_host = self.create_ip_to_host()
+        for member in members:
+            ip_addr = member["addr"].split(":")[0]
+            if rank_0_ip != ip_addr:
+                self.network_down_host = ip_to_host[ip_addr]
+                break
+        self.log.info("network_down_host = %s", self.network_down_host)
+
+        # 2. Create a pool across two ranks on the same node; 0 and rank_r. We have to
+        # provide the size because we're using --ranks.
+        self.add_pool(namespace="/run/pool_size_value/*", create=False)
+        self.pool.target_list.update([0, rank_r])
+        self.pool.create()
+
+        # 3. Create a container without redundancy factor.
+        self.container = []
+        self.container.append(
+            self.get_container(pool=self.pool, namespace="/run/container_wo_rf/*"))
+
+        # 4. Take down the interface where the pool isn't created.
+        errors = []
+        self.interface = self.params.get(
+            "fabric_iface", "/run/server_config/servers/0/*")
+
+        # wolf
+        if self.test_env == "ci":
+            self.update_network_interface(
```

Comment:
(pylint-no-member) Instance of 'NetworkFailureTest' has no 'update_network_interface' member